src/CUAuth/Listeners/AuthorizeUser.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace CornellCustomDev\LaravelStarterKit\CUAuth\Listeners;
+
+use CornellCustomDev\LaravelStarterKit\CUAuth\DataObjects\ShibIdentity;
+use CornellCustomDev\LaravelStarterKit\CUAuth\Events\CUAuthenticated;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Str;
+
+class AuthorizeUser
+{
+    public function handle(CUAuthenticated $event, ?array $serverVars = null): void
+    {
+        $shibboleth = ShibIdentity::fromServerVars($serverVars);
+
+        // Look for a matching user.
+        $userModel = config('auth.providers.users.model');
+        $user = $userModel::firstWhere('email', $shibboleth->email());
+
+        if (empty($user)) {
+            // User does not exist, so create them.
+            $user = new $userModel;
+            $user->name = $shibboleth->name();
+            $user->email = $shibboleth->email();
+            $user->password = Str::random(32);
```

Comment:
I suggest to use eduPersonPrincipalName (netid email) here instead of mail attribute which is alias email. This can be changed later.

Reply:
Apparently I was already in agreement with you! Looking at the ShibIdentity implementation, it uses eduPersonPrincipalName as the value for email if it is available, and it uses mail as a fallback. (See ShibIdentity::fromServerVars() and the phpdoc for ShibIdentity::email())

So, basically, yes, and already done!